app/inpututils.cpp

```diff
@@ -160,6 +160,35 @@ QString InputUtils::formatNumber( const double number, int precision )
   return QString::number( number, 'f', precision );
 }
 
+QString InputUtils::formatDistanceInProjectUnit( const double distanceInMeters, int precision )
+{
+  QgsCoordinateReferenceSystem projectCrs = QgsProject::instance()->crs();
```

Comment:
take CRS from `mapSettings->destinationCrs` similar to `InputUtils::angleBetweenGpsAndFeature`, do not use `instance()` singleton if not really needed.